### ISSUE TITLE
Openstack connector

### ISSUE THREAD

**Issue by FTwex** [silexlabs/unifile#115]
What do you think about using an openstack connector ?

We could build one or use [pkgcloud](https://github.com/pkgcloud/pkgcloud) which provide access to multiple object storage plateform : Amazon, Azure, Google, HP, Openstack and Rackspace.

**Comment by lexoyo** [comment]
a first integration of pkgcloud could be to provide S3 service in unifile

**Comment by JbIPS** [comment]
I'm always open to new connector!

About the lib, its API lacks some functions we'll need to implements Unifile interface (like `.stat()`) but we could use it for the rest of the API.

@FTwex, would you like to try? 😉

**Comment by FTwex** [comment]
Otherwise we may use the official [AWS SDK](https://github.com/aws/aws-sdk-js)
The problem is that it's for all AWS services so it's not lightweight...but with that we have all the methods for Unifile interface (including stats)

**Comment by JbIPS** [comment]
Yes that's another approach. My favorite is the REST API so we have exactly what we want with no extra weight.

**Comment by FTwex** [comment]
Ok they have an [REST API](https://docs.aws.amazon.com/AmazonS3/latest/API/Welcome.html)
I could implement it directly without the sdk

**Comment by FTwex** [comment]
Should I build this new connector as a new package (separate repo like silexlabs/unifile-webdav) or as Unifile lib?

**Comment by JbIPS** [comment]
Since you're the creator of this lib, I think you should keep it under your account.

But please make a PR in Unifile Readme to link to your project 😉

**Comment by JbIPS** [comment]
Hello @FTwex, just checking on you 😃 ! How is it going? Do you need any help?

Let me know if I can be of any assistance.
